app/modules/ads.js

```diff
@@ -349,6 +365,64 @@ class Ads {
 		}
 	}
 
+	isTopLeaderboardApplicable() {
+		const hasFeaturedVideo = this.getTargetingValue('hasFeaturedVideo'),
+			isHome = this.getTargetingValue('pageType') === 'home',
+			$pi = $('.portable-infobox'),
+			$pageHeader = $('.wiki-page-header');
+
+		return isHome || $pi.length > 0 || ($pageHeader.length > 0 && !hasFeaturedVideo);
+	}
+
+	isInContentApplicable() {
+		const $curatedContent = $('.curated-content'),
+			$firstSection = $('.article-content > h2').first(),
+			firstSectionTop = ($firstSection.length && $firstSection.offset().top) || 0;
+
+		if (this.getTargetingValue('pageType') === 'home') {
+			return $curatedContent.length > 0;
+		}
+
+		return firstSectionTop > this.adsData.minZerothSectionLength;
+	}
+
+	isPrefooterApplicable() {
+		const $articleBody = $('.article-body'),
+			$articleFooter = $('.article-footer'),
+			articleBodyHeight = $articleBody.height(),
+			showInContent = this.isInContentApplicable(),
+			$trendingArticles = $('.trending-articles');
+
+		if (this.getTargetingValue('pageType') === 'home') {
```

Comment:
1. :+1:
2. :-1: - we want to return false when it's home and page doesn't have trending articles